mio/process/video.py

```diff
@@ -156,13 +156,28 @@ def process_frame(self, input_frame: np.ndarray) -> Optional[np.ndarray]:
         if input_frame is None:
             return None
 
-        if self.noise_patch_config.enable and self.previous_frame is not None:
-            # Call the unified detection method
-            broken, noise_patch = self.noise_detect_helper.detect_frame_with_noisy_buffer(
+        if self.noise_patch_config.enable:
+            if self.noise_patch_config.method == "gradient":
+            # For the gradient method, analyze the first frame immediately
+                broken, noise_patch = self.noise_detect_helper.detect_frame_with_noisy_buffer(
                 input_frame,
-                self.previous_frame if self.noise_patch_config.method == "mean_error" else None,
+                None,  # No previous frame needed for gradient method
                 self.noise_patch_config,
             )
+            else:
```

Comment:
```suggestion
            if self.noise_patch_config.method == "mean_error":
```
Probably better to make it explicit here.

Reply:
probably with an `elif`

Reply:
oh yes good catch I updated it

Reply:
Oh yes good catch I updated it